TheEngineeringJourney/Assets/Scripts/Map/MapBuilder.cs

```diff
@@ -33,48 +33,76 @@ private void LoadRoomNodeTypeList()
     /// <summary>
     /// Generate random dungeon, returns true if dungeon built, false if failed
     /// </summary>
-    public bool GenerateMap(MapLevelSO currentDungeonLevel)
+    public Build GenerateMap(MapLevelSO currentDungeonLevel)
     {
-        var dungeonBuildSuccessful = false;
-        var dungeonBuildAttempts = 0;
-        var dungeonRebuildAttemptsForNodeGraph = 0;
+        var status = Build.CreateNodeGraph;
+        var buildAttempts = Attempt.Zero;
+        var rebuildAttemptsForNodeGraph = Attempt.Zero;
         var roomTemplateList = currentDungeonLevel.RoomTemplates;
+        // Select a random room node graph from the list
+        var roomNodeGraph = currentDungeonLevel.RoomNodeGraphs.SelectRandomRoomNodeGraph();
         
-        while (!dungeonBuildSuccessful && dungeonBuildAttempts < Settings.MaxMapBuildAttempts)
+        while (true)
         {
-            dungeonBuildAttempts++;
-            // Select a random room node graph from the list
-            var roomNodeGraph = currentDungeonLevel.RoomNodeGraphs.SelectRandomRoomNodeGraph();
-            
-            // Loop until dungeon successfully built or more than max attempts for node graph
-            while (!dungeonBuildSuccessful && dungeonRebuildAttemptsForNodeGraph <= Settings.MaxMapBuildAttempts)
+            switch (status)
             {
-                // Clear dungeon room game objects and dungeon room dictionary
-                ClearMap(MapBuilderRoomDictionary);
-                
-                dungeonRebuildAttemptsForNodeGraph++;
-
-                // Attempt To Build A Random Dungeon For The Selected room node graph
-                dungeonBuildSuccessful = AttemptToBuildRandomMap(roomNodeGraph, MapBuilderRoomDictionary, roomTemplateList, _roomNodeTypeList);
-            }
-
-
-            if (dungeonBuildSuccessful)
-            {
-                // Instantiate Room Game Objects
-                InstantiateRoomGameObjects(MapBuilderRoomDictionary);
+                case Build.CreateNodeGraph:
+                    (status, rebuildAttemptsForNodeGraph) = AttemptCreating(roomNodeGraph, MapBuilderRoomDictionary, roomTemplateList, _roomNodeTypeList, rebuildAttemptsForNodeGraph);
+                    Debug.Log( $"Attempting creating: {status.ToString()}");
+                    break;
+                case Build.ReAttemptFailedCreateNodeGraph:
+                    // Check status 
+                    // If attempt failed then count up 
+                    status = (int)rebuildAttemptsForNodeGraph == Settings.MaxMapBuildAttempts
```

Comment:
I dont like the Settings.MaxMapBuildAttempts is used both in this and attempt failed. How should we do this? Create two parameteres or use the last enum?